src/Middleware/HttpLogging/src/FileLoggerProcessor.cs

```diff
@@ -62,25 +66,38 @@ public FileLoggerProcessor(IOptionsMonitor<W3CLoggerOptions> options, IHostEnvir
             _maxFileSize = loggerOptions.FileSizeLimit;
             _maxRetainedFiles = loggerOptions.RetainedFileCountLimit;
             _flushInterval = loggerOptions.FlushInterval;
+            _fields = loggerOptions.LoggingFields;
             _options.OnChange(options =>
             {
                 lock (_pathLock)
                 {
                     // Clear the cached settings.
                     loggerOptions = options;
+
+                    // Move to a new file if the fields have changed
+                    if (!_fields.Equals(loggerOptions.LoggingFields))
+                    {
+                        _fileNumber++;
+                        if (_fileNumber >= W3CLoggerOptions.MaxFileCount)
+                        {
+                            _maxFilesReached = true;
+                            Log.MaxFilesReached(_logger, new ApplicationException());
+                        }
+                    }
+                    _fields = loggerOptions.LoggingFields;
+
                     if (!string.IsNullOrEmpty(loggerOptions.LogDirectory))
                     {
                         _path = loggerOptions.LogDirectory;
                     }
+
                     _fileName = loggerOptions.FileName;
```

Comment:
If the file name changes should the file number reset, or be rescanned for?

Reply:
Yeah, I think so, that'd be a way for users to get around the 10,000 file a day limit if they wanted to. But it might be somewhat tricky - e.g. OnChange fires in the middle of execution of WriteMessagesAsync, causing the FileNumber to go back to zero. If this happens while we're scanning through some previous files to find out if they exist, it could cause us to suffix the first file with the new name with something nonzero. I'll file a follow-up issue for this one, for now I think it's fine if the numbering for the new file names pick up where the old ones left off.

Reply:
https://github.com/dotnet/aspnetcore/issues/34809